File: 00 Python Programming Fundamentals/Lesson 6 Files/Files.cpp
#include <iostream>
#include <fstream>
using namespace std;
/*
Created by Jonatan Gomez Garcia on 05-01-2024
*/

int main() {
    ifstream fileIn;
    ofstream fileOut;
    string word = "";
    string text = "";

    //Process to read word by word
    fileIn.open("Text to read.txt");
    if (fileIn.is_open()){
        fileIn >> word;
        cout << "Word: " << word << endl;
        fileIn >> word;
        cout << "Word: " << word << endl << endl;
        
    }
    fileIn.close();
    
    //Process to read line by line
    fileIn.open("Text to read.txt");
    if (fileIn.is_open()){
        while(getline(fileIn, text)){
                cout << text <<  endl;
            }
    }

    //Process to write
    fileOut.open("Text to write.txt");
    if (fileOut.is_open()){
        fileOut << "I am writting into the Text to write.txt";
    }
    fileOut.close();


    return 0;
}

File: 00 Python Programming Fundamentals/Lesson 6 Files/Files.py
#!/bin/python3
''' 
Created by Jonatan Gomez Garcia on 05-01-2024
'''

#Read all text
file = open("Text to read.txt", "r")
print(file.read())
file.close()

#Read all text using with (is not necessary to colse file)
with open("Text to read.txt", "r") as f:
    print(f.read())
    print(f.closed)
print(f.closed)

#Read line
file = open("Text to read.txt", "r")
print(file.readline())
file.close()

#Read each line
file = open("Text to read.txt", "r")
for line in file.readlines():
    print(line.replace("\n",""))
file.close()

#Write in file
file = open("Text to write.txt", "w")
file.write("Writting using python.")
file.close()

#Append to file
file = open("Text to write.txt", "a")
file.write("\nGood Morning.")
file.close()
